src/main.rs

```diff
@@ -128,47 +135,35 @@ fn parse_file(filename: &str,
 fn main() {
     env_logger::init().unwrap();
 
-    let args: Vec<String> = env::args().collect();
-    let prog = args[0].clone();
-    let mut options = Options::new();
-    options.optflag("a", "ast", "print AST of input files to STDOUT");
-    options.optmulti("d",
-                     "dot",
-                     "write out GraphViz representations of the input files",
-                     "FILENAME");
-    options.optflag("h", "help", "print this help menu");
-    let matches = match options.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(f) => {
-            usage(prog, f.to_string().as_str(), true, options);
-            process::exit(1);
-        }
-    };
-    debug!("All arguments: {:?}", args);
-    debug!("Free text arguments: {:?}", matches.free);
-    if matches.opt_present("h") {
-        usage(prog, "", false, options);
+    let argv: Vec<String> = env::args().collect();
```

Comment:
Just a FYI: `let argv: Vec<_>` is normally sufficient. `.collect` needs to now what you want to collect into. The ugly alternative is `.collect::<Vec<_>>()`.